packages/earljs/src/isEqual/getCanonicalType.ts

```diff
@@ -0,0 +1,46 @@
+export type CanonicalType = ReturnType<typeof getCanonicalType>
+
+export function getCanonicalType(value: unknown) {
+  const primitive = getBasicType(value)
+  if (primitive) {
+    return primitive
+  } else if (Array.isArray(value)) {
```

Comment:
Why are arrays detected with `isArray` and not with `instanceof Array`?

Reply:
https://web.mit.edu/jwalden/www/isArray.html
What's funny is that the same issue is true for other objects, but only Array.isArray exists. Using Array.isArray ensures correct behavior in loose mode